simavr/sim/avr_uart.c

```diff
@@ -182,7 +200,7 @@ avr_uart_read(
 		goto avr_uart_read_check;
 	}
 	if (!uart_fifo_isempty(&p->input)) { // probably redundant check
-		v = uart_fifo_read(&p->input);
+		v = (uint8_t)uart_fifo_read(&p->input);
```

Comment:
Could you actually use & 0xFF here to clearly represent we're just not typecasting for style?